corehq/ex-submodules/couchforms/getters.py

```diff
@@ -40,13 +43,21 @@ def get_instance_and_attachment(request):
             raise MultimediaBug("Received a submission with POST.keys()")
 
         try:
-            instance = request.FILES[MAGIC_PROPERTY].read()
+            instance_file = request.FILES[MAGIC_PROPERTY]
         except MultiValueDictKeyError:
             instance = MULTIPART_FILENAME_ERROR
         else:
-            for key, item in request.FILES.items():
-                if key != MAGIC_PROPERTY:
-                    attachments[key] = item
+            if not _valid_xml_extension(instance_file):
+                instance = MULTIPART_INVALID_SUBMISSION_FILE_EXTENSION_ERROR
```

Comment:
as mentioned above I think these should raise an exception rather than return a 'fake' instance. I think that would simplify the flow of this function as well.

Reply:
this method is expected to return an instance, so built on top of what already existed. I agree that that can be improved, but chose not to change the core implementation for this change.

Reply:
except it's not returning an instance, it's returning something else. I think what already existed is bad so it'd be good to make it better in the process of making your changes